edb/edgeql/compiler/viewgen.py

```diff
@@ -1534,7 +1534,6 @@ def _normalize_view_ptr_expr(
 
             if (
                 (ctx.expr_exposed or ctx.stmt is ctx.toplevel_stmt)
-                and not qlexpr.limit
```

Comment:
Why is the check for an explicit `limit` on a shape element dropped?

Reply:
Because giving an explicit limit shouldn't override the explicit limit, the minimum should apply.

In most cases, that is what happened with the existing code, since we would typically inject the implicit limit a second time in the inner SelectStmt, but that is no longer the case, so we need to inject here.

Reply:
In theory, one might argue that an explicit LIMIT should fully override an implicit limit, but I don't think that really holds up: there is no way for the CLI/UI to know about the explicit LIMIT, so it will truncate based on the implicit limit anyway.